src/components/Proposal/Create/SchemeForms/Validators.ts

```diff
@@ -0,0 +1,51 @@
+import { isHexStrict } from "web3-utils";
+import { isAddress } from "lib/util";
+
+/**
+ * Given a value returns error message in case value is less than 0 or no value provided
+ * @param {any} value
+ */
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const requireValue = (value: any): string => {
+  let error;
+  if (value === "") {
+    error = "Required";
+  } else if (value < 0) {
+    error = "Please enter a non-negative value";
+  }
+  return error;
+};
+
+/**
+ * Given ABI method param type (address, byets32, unit256, ...) and it's value, returns error message in case validation fails or no value provided.
+ * This function is suitable to work with Formik validations.
+ * @param {string} type
+ * @param {string} value
+ * @returns {undefined | string} undefined if there is no error, otherwise a string representing the error message.
+ */
+export const validateParam = (type: string, value: string): string => {
+  let error;
+  if (!value && type !== "bool") {
+    error = "Required";
+  }
+  else {
+    switch (true) {
```

Comment:
does the switch here is really require ?
how about replacing that to if-else statement  ?

Reply:
Why not? I think it's more elegant in this case. I use `true` since this is the way to use dynamic values in the `case` statements.